server/utils/citations/generateCitationHtml.ts

```diff
@@ -66,8 +66,9 @@ export const generateCitationHtml = async (pubData, communityData) => {
 		type: 'article-journal',
 		title: pubData.title,
 		...authorsEntry,
-		'container-title': communityData.title,
+		'container-title': communityData.citeAs || communityData.title,
```

Comment:
I think I should check what kind of collection this is, but i'm not sure. the same for publisher. that way, this data only gets set for collections that are a book, issue, or conference instead of everything.

Reply:
Yes, this only applies to Journal Issue collections. In Books and Conferences, there is no `container-title`, it's a book or conference title, and we use the Collection's title instead of the Community title. This is because journals have an extra layer to their metadata (Journal Title (Community or Cite As) -> Issue Title (Collection Title) -> Article Title (Pub Title)), whereas books and conference only have two (Book/Conference Title (Collection Title) -> Chapter/Paper Title (Pub Title))

Reply:
Quick question. @3mcd pointed out that the function `getCollectionLevelData` seems to be setting the `container-title` for all collections that are not an issue. This seems to conflict with `container-title` being set only for the issue. Can you clarify is we are interpreting this incorrectly or if this is an issue we should address. 

the function in question
```
const getCollectionLevelData = (collection) => {
	if (!collection) {
		return { type: 'article-journal' };
	}
	const { kind, title } = collection;
	const metadata = collection.metadata || {};
	const useCollectionTitle = collection.kind !== 'issue';
	return {
		type: collectionKindToCitationJSPart(kind),
		...(useCollectionTitle && { 'container-title': title }),
		containerDoi: getCollectionDoi(collection),
		ISBN: metadata.isbn,
		ISSN: metadata.issn || metadata.printIssn || metadata.electronicIssn,
		edition: metadata.edition,
		volume: metadata.volume,
		issue: metadata.issue,
	};
};
```

how it is used
```
	const commonData = {
		// @ts-expect-error ts-migrate(2783) FIXME: 'type' is specified more than once, so this usage ... Remove this comment to see the full error message
		type: 'article-journal',
		title: pubData.title,
		...authorsEntry,
		'container-title': communityData.citeAs || communityData.title,
		...getCollectionLevelData(primaryCollection),
		publisher: communityData.publishAs || '',
	};
```